### ISSUE TITLE
[Bug]: Railway tracks that are perpendicular to shorelines cannot be built on shorelines

### ISSUE THREAD

**Issue by wensimehrp** [OpenTTD/OpenTTD#11771]
### Version of OpenTTD

Master

### Expected result

Railways can be constructed along shorelines, even when they run perpendicular to the coast, similar to the way roads are built.

### Actual result

Railways cannot be built on shorelines if they are perpendicular to the shorelines, leaving an error saying "Cannot build on water".

### Steps to reproduce

1. Start a new game
2. build railways on shorelines

**Comment by krysclarke** [comment]
Please provide a screenshot of the issue.

**Comment by DefinitelyNotRau117** [comment]
Most likely this:
![изображение](https://github.com/OpenTTD/OpenTTD/assets/115672237/3c3b34db-184b-4974-8d7e-8e804bab2f32)

**Comment by WenSimEHRP** [comment]
![OPENTTDBUGTRACK](https://github.com/OpenTTD/OpenTTD/assets/145064583/905f917f-802c-48fb-98d5-ef3cd96405b0)

**Comment by Eddi-z** [comment]
I'm not convinced this needs to be a "bug". Nobody needs a rail at this location, and it would need weird special cases in the code to allow this.

**Comment by 2TallTyler** [comment]
The actual bug is that roads are allowed to do this. Neither should be allowed.

**Comment by LordAro** [comment]
I disagree - the fact that you can build rail while the terrain just happens to not be coast vs when it is coast is weird and unexpected. Introducing this behaviour to roads as well only expands this

I do agree that it's a strange edgecase that is pointless for any real rail construction. But when have we ever stopped players from doing pointless things?

**Comment by LordAro** [comment]
Also, the error message is bad - "cannot build on water" - it's not on water

**Comment by WenSimEHRP** [comment]
> Roads can be built perpendicular to the hillside/coast.
Tracks can be built in any direction on a hillside/coast, except for parts that would have their ending on the lower edge of the foundation

This is what I found on [the wiki.](https://wiki.openttd.org/en/Archive/Manual/Settings/Build%20on%20slopes) I'm not sure if this would help.

**Comment by Eddi-z** [comment]
> The actual bug is that roads are allowed to do this. Neither should be allowed.

Ok, i actually disagree with this. because roads do have a purpose at this location, allowing houses to be built next to them. If there is a need to "harmonize" rail and road construction, it should be on the permissive side.
